dbt_meshify/storage/file_content_editors.py

```diff
@@ -0,0 +1,453 @@
+        updated_yml = self.add_model_version_to_yml(
+            model_name=self.node.name,
+            models_yml=models_yml,
+            prerelease=prerelease,
+            defined_in=defined_in,
+        )
+        # write the updated yml to the file
+        self.file_manager.write_file(yml_path, updated_yml)
+        # create the new version file
+
+        # if we're incrementing the version, write the new version file with a copy of the code
+        latest_version = int(self.node.latest_version) if self.node.latest_version else 0
+        last_version_file_name = f"{self.node.name}_v{latest_version}.{self.node.language}"
+        next_version_file_name = (
+            f"{defined_in}.{self.node.language}"
+            if defined_in
+            else f"{self.node.name}_v{latest_version + 1}.{self.node.language}"
+        )
+        model_path = self.get_resource_path()
+
+        if model_path is None:
+            raise Exception(f"Unable to find path to model {self.node.name}. Aborting.")
+
+        model_folder = model_path.parent
+        next_version_path = model_folder / next_version_file_name
+        last_version_path = model_folder / last_version_file_name
+
+        # if this is the first version, rename the original file to the next version
+        if not self.node.latest_version:
+            Path(self.project_path).joinpath(model_path).rename(
+                Path(self.project_path).joinpath(next_version_path)
+            )
+        else:
+            # if existing versions, create the new one
+            self.file_manager.write_file(next_version_path, self.node.raw_code)
+            # if the existing version doesn't use the _v{version} naming convention, rename it to the previous version
+            if not model_path.root.endswith(f"_v{latest_version}.{self.node.language}"):
+                Path(self.project_path).joinpath(model_path).rename(
+                    Path(self.project_path).joinpath(last_version_path)
+                )
+
+    def update_model_refs(self, model_name: str, project_name: str) -> None:
+        """Updates the model refs in the model's sql file"""
+        model_path = self.get_resource_path()
+
+        if model_path is None:
+            raise Exception(f"Unable to find path to model {self.node.name}. Aborting.")
+
+        # read the model file
+        model_code = str(self.file_manager.read_file(model_path))
+        if self.node.language == "sql":
+            updated_code = self.update_sql_refs(
+                model_name=model_name,
+                project_name=project_name,
+                model_code=model_code,
+            )
+        elif self.node.language == "python":
+            updated_code = self.update_python_refs(
+                model_name=model_name,
+                project_name=project_name,
+                model_code=model_code,
+            )
+        # write the updated model code to the file
+        self.file_manager.write_file(model_path, updated_code)
```

Comment:
Some simplification of our code paths:

```suggestion
        model_code = str(self.file_manager.read_file(model_path))
        # This can be defined in the init for this clas.
        ref_update_methods = {
            'sql': self.update_sql_refs,
            'python': self.update_python_refs
        }
        # Here, we're trusting the dbt-core code to check the languages for us. 🐉 
        updated_code = ref_update_methods[self.node.language](
                model_name=model_name,
                project_name=project_name,
                model_code=model_code,
        )
        # write the updated model code to the file
        self.file_manager.write_file(model_path, updated_code)
```